README.md

```diff
@@ -20,23 +26,100 @@ chel migrate --from <backend> --to <backend> --out <dir-or-sqlitedb>
 
 Note: in many (if not all) instances, the `<url>` parameter can refer to a local folder path, in which case the command will operate without making a network connection, and will instead use the folder's contents to perform its operations.
 
-### `chel keygen`
+### `chel pin` - Per-Contract Versioning System
+
+🎯 Pin individual contracts to specific versions independently!
+
+**Key Features:**
+- ✅ **Per-contract versioning** using `chelonia.json` configuration
+- ✅ **Individual contract pinning** by specifying manifest file path
+- ✅ **New directory structure**: `contracts/<contract-name>/<version>/`
+- ✅ **Manifest-based workflow** - requires existing manifest files
+- ✅ **Ecosystem-agnostic** - no coupling to Node.js/npm
+
+**Workflow:**
+1. **Generate manifest first**: Use `chel manifest` to create `.manifest.json` files
+2. **Pin from manifest**: Use `chel pin` with the manifest file path
+3. **Contract files copied**: Only contract files (main/slim) are copied to new structure
+
+**Usage Examples:**
+```bash
+chel pin <version> <manifest-file-path>
+
+# Pin specific contract to a version using its manifest (from dist/contracts)
+chel pin 2.0.5 dist/contracts/2.0.5/chatroom.2.0.5.manifest.json
+chel pin 2.0.0 dist/contracts/2.0.0/group.2.0.0.manifest.json
 
+# Switch versions efficiently with --only-changed
+chel pin 2.0.6 dist/contracts/2.0.6/chatroom.2.0.6.manifest.json --only-changed
+
+# Note: Contracts are pinned to the contracts/ output directory
 ```
+
+**Configuration (`chelonia.json`):**
+```json
 {
-  "version": "1.0.0",
-  "cipher": "algo",
-  "pubkey": "...",
-  "privkey": "...",
-  "encrypted": "algo"
+  "contracts": {
+    "chatroom": {
+      "version": "2.0.6",
+      "path": "contracts/gi.contracts_chatroom/2.0.6/chatroom.2.0.6.manifest.json"
+    },
+    "group": {
+      "version": "2.0.0",
+      "path": "contracts/gi.contracts_group/2.0.0/group.2.0.0.manifest.json"
+    }
+  }
 }
 ```
 
-If `"encrypted"` doesn't exist - it means the `"privkey"` was saved in the clear.
+**Directory Structure Created:**
+```
+contracts/
+├── gi.contracts_chatroom/
+│   ├── 2.0.5/
+│   │   ├── chatroom.js
+│   │   └── chatroom-slim.js
+│   └── 2.0.6/
+│       ├── chatroom.js
+│       └── chatroom-slim.js
+└── gi.contracts_group/
+    └── 2.0.0/
+        ├── group.js
+        └── group-slim.js
+```
+
+> **Note:** The pin command only copies contract files (main and slim) from existing manifest files. It does not copy or generate manifest files - those should be created separately using `chel manifest`.
```

Comment:
This is a mistake. Manifest files should definitely be copied. Contracts are referenced from Chelonia by the hash of the manifest files after all, so without them you can't load contracts at all in a Chelonia app.

Reply:
This is `chel manifest` btw. 
IIRC @corrideat suggested not to copy manifest files in `chel pin`.

Reply:
@akhileshthite Do you know which comment you're referring to? The closest I can find is https://github.com/okTurtles/chel/pull/64#discussion_r2296270349, but there I'm not saying that manifest files shouldn't be copied over, just that the logic for copying should take versions into account.

Reply:
Replied on slack.